contracts/USDCFlow.cdc

```diff
@@ -18,159 +20,207 @@ import "FiatToken"
-                        ftDisplay: self.resolveView(Type<FungibleTokenMetadataViews.FTDisplay>()) as! FungibleTokenMetadataViews.FTDisplay?,
-                        ftVaultData: self.resolveView(Type<FungibleTokenMetadataViews.FTVaultData>()) as! FungibleTokenMetadataViews.FTVaultData?
-                    )
-                case Type<FungibleTokenMetadataViews.FTDisplay>():
-                    let media = MetadataViews.Media(
-                            file: MetadataViews.HTTPFile(
-                            url: "https://assets.website-files.com/5f6294c0c7a8cdd643b1c820/5f6294c0c7a8cda55cb1c936_Flow_Wordmark.svg"
-                        ),
-                        mediaType: "image/svg+xml"
-                    )
-                    let medias = MetadataViews.Medias([media])
-                    return FungibleTokenMetadataViews.FTDisplay(
-                        name: "USDC (Flow)",
-                        symbol: "USDCf",
-                        description: "This fungible token representation of USDC is bridged from Flow EVM.",
-                        externalURL: MetadataViews.ExternalURL("https://www.circle.com/en/usdc"),
-                        logos: medias,
-                        socials: {
-                            "x": MetadataViews.ExternalURL("https://x.com/circle")
-                        }
-                    )
-                case Type<FungibleTokenMetadataViews.FTVaultData>():
-                    return FungibleTokenMetadataViews.FTVaultData(
-                        storagePath: USDCFlow.VaultStoragePath,
-                        receiverPath: USDCFlow.ReceiverPublicPath,
-                        metadataPath: USDCFlow.VaultPublicPath,
-                        providerPath: /private/usdcFlowVault,
-                        receiverLinkedType: Type<&USDCFlow.Vault{FungibleToken.Receiver}>(),
-                        metadataLinkedType: Type<&USDCFlow.Vault{FungibleToken.Balance, MetadataViews.Resolver}>(),
-                        providerLinkedType: Type<&USDCFlow.Vault{FungibleToken.Provider}>(),
-                        createEmptyVaultFunction: (fun (): @USDCFlow.Vault {
-                            return <-USDCFlow.createEmptyVault()
-                        })
-                    )
-            }
-            return nil
+        access(all) fun resolveView(_ view: Type): AnyStruct? {
+            return USDCFlow.resolveContractView(resourceType: nil, viewType: view)
         }
     }
 
-    /// createEmptyVault
-    ///
-    /// @return The new Vault resource with a balance of zero
-    ///
-    pub fun createEmptyVault(): @Vault {
-        return <-create Vault(balance: 0.0)
-    }
+    access(all) resource Minter: FlowEVMBridgeHandlerInterfaces.TokenMinter {
 
-    /// wrapFiatToken
-    ///
-    /// Provides a way for users to exchange a FiatToken Vault
-    /// for a USDCFlow Vault with the same balance
-    pub fun wrapFiatToken(_ from: @FungibleToken.Vault): @Vault {
```

Comment:
Are we no longer allowing users to wrap FiatToken to USDCFlow come Cadence 1.0? I thought one of the main goals was to have a redemption path for existing FiatToken Vaults via wrapping their old vault, but I could have misunderstood the aim

Reply:
That was never the plan. Circle is doing redemptions via a snapshot of balances at the time of the upgrade, so allowing wrapping after Crescendo would enable double spending.

Reply:
Got it, thanks for clarifying